org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2010-2022 BSI Business Systems Integration AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ */
+package org.eclipse.scout.rt.dataobject.id;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.exception.PlatformException;
+import org.eclipse.scout.rt.platform.util.LazyValue;
+import org.eclipse.scout.rt.platform.util.StringUtility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Codec used to convert bewteen {@link IId} instances and their qualified/unqualified representation as {@link String}.
+ */
+@ApplicationScoped
+public class IdCodec {
+
+  private static final Logger LOG = LoggerFactory.getLogger(IdCodec.class);
+
+  protected final LazyValue<IdFactory> m_idFactory = new LazyValue<>(IdFactory.class);
+  protected final LazyValue<IdInventory> m_idInventory = new LazyValue<>(IdInventory.class);
+
+  protected final Map<Class<?>, Function<String, Object>> m_rawTypeFromStringMapper = new HashMap<>();
```

Comment:
Could we have threading issues here because maps might be populated by calls to registerRawTypeMapper auf class initialization? E.g. when there are already read accesses. If protected method is only designed to be used when class is replaced and initialize is overriden, thus is fine. Maybe state as JavaDoc on registerRawTypeMapper.